server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1405,35 +1891,39 @@ public Pool createPool(@PathParam("owner_key") @Verify(Owner.class) String owner
         value = "Update Pool")
     @ApiResponses({ @ApiResponse(code = 404, message = "Owner not found") })
     public void updatePool(@PathParam("owner_key") @Verify(Owner.class) String ownerKey,
-        @ApiParam(name = "pool", required = true) Pool newPool) {
+        @ApiParam(name = "pool", required = true) PoolDTO newPoolDTO) {
 
-        Pool currentPool = this.poolManager.find(newPool.getId());
+        Pool currentPool = this.poolManager.find(newPoolDTO.getId());
         if (currentPool == null) {
             throw new NotFoundException(
-                i18n.tr("Unable to find a pool with the ID \"{0}\"", newPool.getId()));
+                i18n.tr("Unable to find a pool with the ID \"{0}\"", newPoolDTO.getId()));
         }
 
         // Verify the existing pool belongs to the specified owner
         // If the pool isn't valid for this owner, we'll pretend it doesn't exist to avoid any
         // potential security concerns.
         if (currentPool.getOwner() == null || !ownerKey.equals(currentPool.getOwner().getKey())) {
             throw new NotFoundException(
-                i18n.tr("Unable to find a pool with the ID \"{0}\"", newPool.getId()));
+                i18n.tr("Unable to find a pool with the ID \"{0}\"", newPoolDTO.getId()));
         }
 
-        // Ignore what the client has specified as the owner, since we already know the owner here
-        newPool.setOwner(currentPool.getOwner());
-
         // Verify the pool type is one that allows modifications
-        if (currentPool.getType() != PoolType.NORMAL || newPool.getType() != PoolType.NORMAL) {
+        if (currentPool.getType() != PoolType.NORMAL ||
+            !newPoolDTO.getType().equals(PoolType.NORMAL.toString())) {
             throw new BadRequestException(i18n.tr("Cannot update bonus pools, as they are auto generated"));
         }
 
-        if (currentPool.isCreatedByShare() || newPool.isCreatedByShare()) {
+        if (currentPool.isCreatedByShare() || newPoolDTO.isCreatedByShare()) {
```

Comment:
We're not interested in this value on the DTO -- only use the actual entity as the source of truth for state validation.